Add more linker hints for esp-radio items

For example, if users don't have esp-rtos in their project and try to use esp-radio they might get linker errors like 

```
3a9594f9cc748595-cgu.0.rcgu.o:(.literal._ZN9esp_radio6compat5mutex12mutex_create17h8d23aa1a4e836e60E+0x0): undefined reference to `esp_rtos_semaphore_create'
```

--- a/template/build.rs
+++ b/template/build.rs
@@ -18,7 +18,7 @@ fn linker_be_nice() {
 
         match kind.as_str() {
             "undefined-symbol" => match what.as_str() {
-                kind if kind.starts_with("_defmt_") => {
+                what if what.starts_with("_defmt_") => {
                     eprintln!();
                     eprintln!(
                         "💡 `defmt` not found - make sure `defmt.x` is added as a linker script and you have included `use defmt_rtt as _;`"
@@ -30,7 +30,7 @@ fn linker_be_nice() {
                     eprintln!("💡 Is the linker script `linkall.x` missing?");
                     eprintln!();
                 }
-                kind if kind.starts_with("esp_rtos_") => {
+                what if what.starts_with("esp_rtos_") => {
                     eprintln!();
                     eprintln!(
                         "💡 `esp-radio` has no scheduler enabled. Make sure you have initialized `esp-rtos` or provided an external scheduler."

